### PR TITLE
k8s: Support circular owner references

### PR DESCRIPTION
**Pull request by mterwill**
We introduced a circular owner reference (so one CRD gets cleaned up when another is deleted and vice versa) and found that Tilt got stuck in pending for all resources. I peeked at pprof and this looked suspicious:

```
25 @ 0x104a19594 0x1049e2d1c 0x1049e2884 0x106252da8 0x106252da9 0x106253908 0x106252d30 0x106253908 0x1062534c0 0x106d6c818 0x104a4f494
#	0x106252da7	github.com/tilt-dev/tilt/internal/k8s.(*objectTreePromise).wait+0x2e7						/root/project/internal/k8s/owner_fetcher.go:275
#	0x106252da8	github.com/tilt-dev/tilt/internal/k8s.OwnerFetcher.OwnerTreeOfRef+0x2e8						/root/project/internal/k8s/owner_fetcher.go:157
#	0x106253907	github.com/tilt-dev/tilt/internal/k8s.OwnerFetcher.ownerTreeOfHelper+0x217					/root/project/internal/k8s/owner_fetcher.go:222
#	0x106252d2f	github.com/tilt-dev/tilt/internal/k8s.OwnerFetcher.OwnerTreeOfRef+0x26f						/root/project/internal/k8s/owner_fetcher.go:175
#	0x106253907	github.com/tilt-dev/tilt/internal/k8s.OwnerFetcher.ownerTreeOfHelper+0x217					/root/project/internal/k8s/owner_fetcher.go:222
#	0x1062534bf	github.com/tilt-dev/tilt/internal/k8s.OwnerFetcher.OwnerTreeOf+0x26f						/root/project/internal/k8s/owner_fetcher.go:214
#	0x106d6c817	github.com/tilt-dev/tilt/internal/controllers/core/kubernetesdiscovery.(*Reconciler).handlePodChange+0xb7	/root/project/internal/controllers/core/kubernetesdiscovery/reconciler.go:628
```

Found that Tilt was deadlocked on these promises for resources that had owner references pointing to one another.

### CODE REVIEW

File: internal/k8s/owner_fetcher.go [hunk 1]
```diff
@@ -211,15 +216,20 @@ func (v OwnerFetcher) OwnerTreeOf(ctx context.Context, entity K8sEntity) (result
...
-	return v.ownerTreeOfHelper(ctx, ref, meta)
+	return v.ownerTreeOfHelper(ctx, ref, meta, nil)
 }
 
-func (v OwnerFetcher) ownerTreeOfHelper(ctx context.Context, ref v1.ObjectReference, meta metav1.Object) (ObjectRefTree, error) {
+func (v OwnerFetcher) ownerTreeOfHelper(ctx context.Context, ref v1.ObjectReference, meta metav1.Object, path []types.UID) (ObjectRefTree, error) {
 	tree := ObjectRefTree{Ref: ref, CreationTimestamp: meta.GetCreationTimestamp()}
 	owners := meta.GetOwnerReferences()
 	for _, owner := range owners {
+		// TODO: Owner references can also exist at cluster scope, for which this incorrectly propagates the parent ref's Namespace.
```
Comment (mterwill) [inline internal/k8s/owner_fetcher.go:226]: Not sure what you want to do here. I noticed this as I was testing but it doesn't seem to materially change how Tilt operates and as far as I could tell there's no easy way in this code path to determine if a resource exists at cluster scope. The namespace of the owner reference is not specified in its object ([ref](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#OwnerReference)).

> OwnerReference contains enough information to let you identify an owning object. An owning object must be in the same namespace as the dependent, or be cluster-scoped, so there is no namespace field. +structType=atomic

Comment (nicks) [reply internal/k8s/owner_fetcher.go:226]: maybe it's ok? our client library wrappers all require a namespace argument, but i think they just drop it if the resource is cluster-scoped.

**Review by nicks — COMMENTED** [review]
ha! today i learned that owner refs could be circular!

i think the test failure on ci looks legit?

**Comment by mterwill** [comment]
Yes the test failure was definitely legit! I was originally using a slice of the entire owner reference (rather than just the UID) which is how I realized the namespace discrepancy. Was appending the wrong UID to `path` when I switched it, thanks test 😅

File: internal/k8s/owner_fetcher.go (hunk 1 above)
Comment (mterwill) [reply internal/k8s/owner_fetcher.go:226]: Want me to adjust to a `NOTE` or anything?

**Comment by mterwill** [comment]
OK now I think this test failure might be a flake?

**Comment by nicks** [comment]
yep! merging

**Review by nicks — APPROVED** [review]

**Pull request merged**